### PR TITLE
Refactor/my addresses #296

### LINKED ISSUE #296: Styling of My Addresses / My ORders / Use Storefront UI components

### PR DESCRIPTION
**Pull request by mkucmus**
closes #296 
- list of all addresses
- mark default one with special SfIcon
- select other address and make it default one
- allow deleting an address

### CODE REVIEW

File: packages/composables/src/hooks/useUser.ts [hunk 1]
```diff
@@ -109,7 +124,55 @@ export const useUser = (): UseUser => {
...
+  }: {
+    addressId?: string;
+    type?: AddressType;
+  }): Promise<boolean> => {
+    if (!addressId || !type) {
+      return false;
+    }
+
+    try {
+      switch (type) {
```
Comment (krskibin) [inline packages/composables/src/hooks/useUser.ts]:
```suggestion
     if (type === AddressType.billing) {   // or even "billing"
       await setDefaultCustomerBillingAddress(addressId);
     } else {
       await setDefaultCustomerShippingAddress(addressId);
     }
     await refreshUser();
     return true;
```

File: packages/default-theme/components/account/MyAddresses.vue [hunk 2]
```diff
@@ -0,0 +1,86 @@
+<template>
+  <div class="shipping-list">
+      <Address v-for="address in addresses"
+      :address="address"
+      :key="address.id"
+      :isDefaultShipping="address.id === selectedShipping"
+      :isDefaultBilling="address.id === selectedBilling"
+      v-on:selectDefaultAddress="selectDefaultAddress"
+      v-on:deleteAddress="deleteAddress"
+      class="shipping"/>
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses.vue]:
```suggestion
      class="shipping-list__address" />
```

File: packages/default-theme/components/account/MyAddresses.vue [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    }
+  },
+  async mounted() {
+     await this.loadAddresses();
+  },
+  computed: {
+  },
+  methods: {
+    async selectDefaultAddress(addressId, type) {
+      console.warn('selectDefaultAddress', addressId, type);
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses.vue]:
```suggestion

```

File: packages/default-theme/components/account/MyAddresses.vue [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      await this.markAddressAsDefault({addressId, type})
+      switch(type) {
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses.vue]:
```suggestion
      if (type === 'shipping') {
        this.selectedShipping = addressId
      } else 
      // ...
```

File: packages/default-theme/components/account/MyAddresses.vue [hunk 5]
```diff
@@ -0,0 +1,86 @@
+<template>
+  <div class="shipping-list">
+      <Address v-for="address in addresses"
+      :address="address"
+      :key="address.id"
+      :isDefaultShipping="address.id === selectedShipping"
+      :isDefaultBilling="address.id === selectedBilling"
+      v-on:selectDefaultAddress="selectDefaultAddress"
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses.vue]: There is a shortcut for `v-on:`, `@selectDefaultAddress`

File: packages/default-theme/components/account/MyAddresses.vue [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    }
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '~@storefront-ui/vue/styles.scss';
+@import '~@storefront-ui/shared/styles/helpers/visibility';
+
+.shipping-list {
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses.vue]:
```suggestion
.shipping-list {
  margin-bottom: $spacer-extra-big;
  width: 100%;
  &__address {
    display: flex;
    padding: $spacer-big 0;
    border-top: 1px solid $c-light;
  }
}
```

File: packages/default-theme/components/account/MyAddresses/Address.vue [hunk 7]
```diff
@@ -0,0 +1,178 @@
+<template>
+<div v-if="address">
+  <div class="shipping__content">
+  <p class="shipping__address">
+    <span class="shipping__client-name">
+      {{ firstName }} {{ lastName }}
+    </span>
+    <br />
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses/Address.vue]: Please, don't use `<br>` tag

File: packages/default-theme/components/account/MyAddresses.vue [hunk 8]
```diff
@@ -0,0 +1,86 @@
+<template>
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses.vue]: There is no possibility to add a new address.

File: packages/default-theme/components/account/MyAddresses/Address.vue [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+      />
+      <span v-if="isDefaultBilling" class="mobile-only">default billing address</span>
+    <SfIcon
+      icon="cross"
+      :color="isDefaultBilling || isDefaultShipping ? 'grey' : 'pink-primary'"
+      size="xs"
+      role="button"
+      title="Delete"
+      v-on:click="(isDefaultBilling || isDefaultShipping) || $emit('deleteAddress', address.id)"
+      :class="(isDefaultBilling || isDefaultShipping) && 'info'"
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses/Address.vue]:
```suggestion
      :class="{ info: isDefaultBilling || isDefaultShipping }"
```
or something like that

File: packages/default-theme/components/account/MyAddresses/Address.vue [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+import { SfProperty, SfTabs, SfList, SfButton, SfIcon, SfBadge, SfCheckbox } from '@storefront-ui/vue'
+import { useUser } from '@shopware-pwa/composables'
+
+export default {
+  name: "Address",
+  components: {SfProperty, SfTabs, SfList, SfButton, SfIcon, SfBadge, SfCheckbox},
+  props: {
+    address: {
+      type: Object,
+      required: true
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses/Address.vue]:
```suggestion
      default: () => {}
```

File: packages/default-theme/components/account/MyAddresses/Address.vue [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+    },
+    isDefaultBilling: {
+      type: Boolean,
+      required: false
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses/Address.vue]:
```suggestion

```

File: packages/default-theme/components/account/MyAddresses/Address.vue [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+    },
+    isDefaultShipping: {
+      type: Boolean,
+      required: false
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses/Address.vue]:
```suggestion

```

File: packages/default-theme/components/account/MyAddresses/Address.vue [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+    }
+  },
+  data() {
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses/Address.vue]: unnecessary added to component
```suggestion
```

File: packages/default-theme/components/account/MyAddresses/Address.vue [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+      return this.address.zipcode
+    },
+    phone() {
+      return this.address.phoneNumber
+    },
+    country() {
+      return this.address.country && address.country.name
+    }
+  },
+  methods: {
```
Comment (krskibin) [inline packages/default-theme/components/account/MyAddresses/Address.vue]: here too:
```suggestion

```

File: packages/default-theme/components/account/MyAddresses/Address.vue (hunk 7 above)
Comment (mkucmus) [reply packages/default-theme/components/account/MyAddresses/Address.vue]: you can make it on your own ;) I just took it from Ui example so it's always better to have it done than perfect :+1:

File: packages/default-theme/components/account/MyAddresses.vue (hunk 5 above)
Comment (mkucmus) [reply packages/default-theme/components/account/MyAddresses.vue]: nice :+1: changing it!

File: packages/default-theme/components/account/MyAddresses/Address.vue (hunk 9 above)
Comment (mkucmus) [reply packages/default-theme/components/account/MyAddresses/Address.vue]: :+1:

File: packages/default-theme/components/account/MyAddresses/Address.vue (hunk 14 above)
Comment (mkucmus) [reply packages/default-theme/components/account/MyAddresses/Address.vue]: :+1:

File: packages/default-theme/components/account/MyAddresses/Address.vue (hunk 13 above)
Comment (mkucmus) [reply packages/default-theme/components/account/MyAddresses/Address.vue]: :+1:

File: packages/default-theme/components/account/MyAddresses/Address.vue (hunk 12 above)
Comment (mkucmus) [reply packages/default-theme/components/account/MyAddresses/Address.vue]: :+1:

File: packages/composables/src/hooks/useUser.ts (hunk 1 above)
Comment (mkucmus) [reply packages/composables/src/hooks/useUser.ts]: I made a little refactor down here :+1:  I've left a switch, in this case, is more readable. 
I rather use if -else for boolean values, for data, I would stay with the switch, which is also even faster than ifs ;-)

File: packages/default-theme/components/account/MyAddresses.vue (hunk 8 above)
Comment (mkucmus) [reply packages/default-theme/components/account/MyAddresses.vue]: yes, it's true. It wasn't planned to handle the forms. We need your powers here ;)

**Pull request merged**